test-util/src/main/java/io/camunda/zeebe/test/util/bpmn/random/ExecutionPathSegment.java

```diff
@@ -259,6 +281,14 @@ public String toString() {
     return ToStringBuilder.reflectionToString(this, ToStringStyle.SHORT_PREFIX_STYLE);
   }
 
+  public boolean hasReachedTerminateEndEvent() {
+    return reachedTerminateEndEvent;
+  }
+
+  public void setReachedTerminateEndEvent(final boolean reachedTerminateEndEvent) {
+    this.reachedTerminateEndEvent = reachedTerminateEndEvent;
+  }
```

Comment:
I don't know if Java has something like that. But an option is changing the method to:
```java
public void setReachedTerminateEndEvent() {
    this.reachedTerminateEndEvent = true;
  }
```
Don't really need a Java specific one-way boolean for this 😄 

Now unfortunately this doesn't work here as it's not always set to true. In some cases a Random decides whether or not it should be true:
```java
result.setReachedTerminateEndEvent(boundaryEventBuilder.timerEventHasTerminateEndEvent());
```
Since we don't want to do an if-check everywhere it's simpler to just use an ordinary setter.

Reply:
You can use the `not` operator `!`. 

https://www.baeldung.com/java-toggle-boolean#toggling-a-primitive-boolean-variable

Reply:
@Zelldon Interesting operator, but this allows toggling both ways. It's also just an operator and not a type that restricts how the value can be altered. So it's not what I was looking for. In Java, we'd need some class that restricts both the usage (you can't just set any value) and the behavior (if you try to change it, it will only ever change once, i.e. one-way switch). 

@remcowesterhoud Why does this have to be set from outside? Can't the path keep track of this value itself, e.g. when append is called?

Reply:
Ok than I got your question wrong. But what about atomic boolean and cas the value? :thinking:

Reply:
> @remcowesterhoud Why does this have to be set from outside? Can't the path keep track of this value itself, e.g. when append is called?

We do set this as part of the append method. So it partially does keep track of it itself. We still need the setter.

Imagine we have execution path segment A, and B. Segment A has not reached a terminate end event. Segment B has reached a terminate end event. If I append B to A, A will check if the path that's being added has reached a terminate end event. This is the case here. So long as A doesn't change the flow scope it means that A will now have reached a terminate end event as well.

The problem is that we need to know that B has reached this terminate end event. Nothing is appended to B at this point. You basically want to set this terminate flag at construction time. Unfortunately, when we create this segment we don't know whether or not it will reach a terminate end event. This depends on some randoms that happen afterwards. As a result the only reason around this (that I could think of) is using a setter to flip this flag.